sw/airborne/modules/osd_max7456_new/max7456.c

```diff
@@ -0,0 +1,1364 @@
+max7456_osd_status = OSD_UNINIT;
+step = 0;
+
+return;
+}
+
+//222222222222222222222222222222222222222222222222222222222222222222222222222222222222222222
+void max7456_periodic(void){
+
+//This code is executed always and checks if the "osd_enable" var has been changed by telemetry.
+//If yes then it commands a reset but this time turns on or off the osd overlay, not the video. 
+if (max7456_osd_status == OSD_IDLE){
+   if(osd_enable > 1){ osd_enable = 1; }
+   if ((osd_enable<<3) !=  osd_enable_val){
+      osd_enable_val = (osd_enable<<3);
+      max7456_osd_status = OSD_UNINIT;
+   }
+}
+
+//INITIALIZATION OF THE OSD
+if (max7456_osd_status == OSD_UNINIT){
+   step = 0;
+   max7456_trans.status = SPITransDone;
+   max7456_trans.output_buf[0] = OSD_VM0_REG;
+   //This operation needs at least 100us but when the periodic function will be invoked again
+   //sufficient time will have elapsed even with at a periodic frequency of 1000 Hz 
+   max7456_trans.output_buf[1] = OSD_RESET; 
+   max7456_osd_status = OSD_INIT1;
+   spi_submit(&(MAX7456_SPI_DEV), &max7456_trans);
+}
+else
+if (max7456_osd_status == OSD_INIT2){
+   max7456_trans.output_length = 1;
+   max7456_trans.input_length = 1;
+   max7456_trans.output_buf[0] = OSD_OSDBL_REG_R;
+   max7456_osd_status = OSD_INIT3;
+   spi_submit(&(MAX7456_SPI_DEV), &max7456_trans); 
+}
+else
+if (max7456_osd_status == OSD_IDLE && osd_enable > 0){ // DRAW THE OSD SCREEN
+   
+   draw_osd_flying_screen();
+
+} // if (max7456_osd_status == OSD_IDLE && osd_enable > 0)
+
+#if OSD_USE_MAG_COMPASS && !defined(SITL)
+
+#ifndef AHRS_MAG_DECLINATION
+#define AHRS_MAG_DECLINATION 5.3 //Sept 2020 for Greece
+#endif
+float mag_heading_x = 0;
+float mag_heading_y = 0;
+
+MAG_Heading = 0;
+
+
+int32_t x = 0, y = 0, z = 0;
+
+x = (imu.mag_unscaled.x - imu.mag_neutral.x)*IMU_MAG_X_SIGN;
+y = (imu.mag_unscaled.y - imu.mag_neutral.y)*IMU_MAG_Y_SIGN;
+z = (imu.mag_unscaled.z - imu.mag_neutral.z)*IMU_MAG_Z_SIGN;
+
+float cos_roll; float sin_roll; float cos_pitch; float sin_pitch;
+struct FloatEulers* att = stateGetNedToBodyEulers_f();
+cos_roll = cosf(att->phi);
+sin_roll = sinf(att->phi);
+cos_pitch = cosf(att->theta);
+sin_pitch = sinf(att->theta);
+// Pitch&Roll Compensation:
+mag_heading_x = x*cos_pitch + y*sin_roll*sin_pitch + z*cos_roll*sin_pitch;
+mag_heading_y = y*cos_roll - z*sin_roll;
+// Magnetic Heading
+MAG_Heading = atan2(-mag_heading_y,mag_heading_x);
+//compass = MAG_Heading;
+// MAG_Heading = atan2(imu.mag.y, -imu.mag.x); // Magnetic Heading 2D
+// Declination correction (if supplied)
+if ( AHRS_MAG_DECLINATION != 0.0 ){
+   MAG_Heading = MAG_Heading + AHRS_MAG_DECLINATION;
+   if (MAG_Heading > M_PI){  // Angle normalization (-180 deg to 180 deg)
+      MAG_Heading -= (2.0 * M_PI);
+
+   }else if (MAG_Heading < -M_PI){ MAG_Heading += (2.0 * M_PI); }
+}
+#endif
+return;
+}
+
+//333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333
+void max7456_event(void){ 
```

Comment:
you should run the `./fix_code_style.sh` script on this file to have the indentation recommended for Paparazzi